client/src/pages/Medical.js

```diff
@@ -1,52 +1,62 @@
-import axios from 'axios';
-import React, { useState, useEffect } from 'react';
-import { Nav } from 'react-bootstrap';
+import axios from "axios";
+import React, { useState, useEffect } from "react";
+import { Nav } from "react-bootstrap";
 
-import { getLocalStorageJson } from '../utils/local-storage';
+import { getLocalStorageJson } from "../utils/local-storage";
 
 const CONTAINER_STYLES = {
-    marginTop: "30px",
-    width: "600px",
-    flexDirection: 'column',
-    alignItems: 'stretch',
-    justifyContent: 'space-between',
-}
+  marginTop: "30px",
+  width: "600px",
+  flexDirection: "column",
+  alignItems: "stretch",
+  justifyContent: "space-between"
+};
 
 const INITIAL_STATE = {
-    emergencyNumber: '',
-    country: '',
+  emergencyNumber: "",
+  country: ""
 };
 
-const needHelpAnswers = getLocalStorageJson('needHelpAnswers') || [];
-const geolocation = needHelpAnswers.find(answer => Object.keys(answer).includes('location')).location;
+const needHelpAnswers = getLocalStorageJson("needHelpAnswers") || [];
+
+const getGeoLocation = () => {
+  if (needHelpAnswers && needHelpAnswers.length) {
+    return needHelpAnswers.find(answer =>
+      Object.keys(answer).includes("location")
+    ).location;
+  }
+  return undefined;
+};
 
 export const Medical = () => {
-    const [state, setState] = useState(INITIAL_STATE);
-    useEffect(() => {
-        (async function fetchData() {
-            const result = await axios.post('/api/geo/country', geolocation );
-            setState({ ...state, country: result.data });
-        })();
-    }, [geolocation]);
-
-    return (
-        <div className="text-center mx-auto" style={ CONTAINER_STYLES }>
-            <h5>Local Emergency Number</h5>
-            <h1 className="text-primary display-4 font-weight-bolder">911</h1>
-
-            <div style={ { display: "flex", margin: "30px 10px 10px 50px" } }>
-                <Nav variant="med-info">
-                    <Nav.Item>
-                        <Nav.Link>Nearest Hospitals</Nav.Link>
-                    </Nav.Item>
-                    <Nav.Item>
-                        <Nav.Link>Symptoms Check</Nav.Link>
-                    </Nav.Item>
-                    <Nav.Item>
-                        <Nav.Link>Find Help</Nav.Link>
-                    </Nav.Item>
-                </Nav>
-            </div>
-        </div>
-    );
+  const [state, setState] = useState(INITIAL_STATE);
+  const fetchData = () => {
+    const geolocation = getGeoLocation();
+    axios
+      .post("/api/geo/country", geolocation)
+      .then(res => setState({ ...state, country: res.data }))
+      .catch(err => console.log(err));
+  };
+  useEffect(fetchData, []);
```

Comment:
I think this dependency of the hook was correct, because if `geolocation` changes the hook needs to execute again to refetch the country